pkg/kubelet/kubelet_pods.go

```diff
@@ -472,11 +472,17 @@ func (kl *Kubelet) GenerateRunContainerOptions(pod *v1.Pod, container *v1.Contai
 			glog.Errorf("Error on creating %q: %v", p, err)
 		} else {
 			opts.PodContainerDir = p
+			if err := kl.chownDirForRemappedIDs(p); err != nil {
```

Comment:
I expected some information about the container to be input to this chown method... can you describe what is being done here?

Reply:
container information is not being passed because non-zero ID on host mapped with 0 ID on container is same for all the containers on a node. And we are using only this non-zero ID to chown the paths which are otherwise owned by ID 0 of host.

Reply:
is this chown happening for volumes mounted or just the container file system ?

Reply:
for the volumes as well. But only for the volumes in var/lib/kubelet/pods/pod-id/volumes/ and not for remote volumes.